storage/remote/queue_manager.go

```diff
@@ -1452,40 +1542,50 @@ func (s *shards) populateTimeSeries(batch []timeSeries, pendingData []prompb.Tim
 		case tFloatHistogram:
 			pendingData[nPending].Histograms = append(pendingData[nPending].Histograms, FloatHistogramToHistogramProto(d.timestamp, d.floatHistogram))
 			nPendingHistograms++
+		case tMetadata:
+			pendingData[nPending].Metadata = prompb.Metadata{
+				Type: metricTypeToProtoEquivalent(d.metadata.Type),
+				Help: d.metadata.Help,
+				Unit: d.metadata.Unit,
+			}
+			nPendingMetadata++
```

Comment:
Hm, so this wouldn't make metadata available inline for series with samples/exemplars/histograms/etc. It would be sent entirely separately in the array of `timeseries` sent and the labels would be duplicated inside a request where samples/other were sent for a timeseries.

Currently it looks like the metadata record is enqueued separately and therefore would never have samples/exemplars/histograms/etc set for this entry in `pendingData[nPending]`:
https://github.com/prometheus/prometheus/blob/208922bdb712224075c44aebe7c6cad7a194c56b/storage/remote/queue_manager.go#L817-L828

Previously with this working (in PR https://github.com/prometheus/prometheus/pull/7771) end to end instead we kept the metadata available then attached it to samples as they were enqueued:
https://github.com/prometheus/prometheus/blob/b6e5bd13405764da8ba04e7ae66f02b877c44187/storage/remote/queue_manager.go#L341-L367

Then this was set inline when the timeseries with samples entry was sent:
https://github.com/prometheus/prometheus/blob/b6e5bd13405764da8ba04e7ae66f02b877c44187/storage/remote/queue_manager.go#L864-L866